### ISSUE TITLE
enhancement, displaying our contacts on the map

### ISSUE THREAD

**Issue by promeneur** [osmandapp/Osmand#5699]
osmand+ 3.0.4

it would be a handy feature: an option to display our contacts over the map

**Comment by carstenhag** [comment]
I would also really like this. Google Maps shows a label on the map for a contact when the Adress is given. Would be pretty handsome to have these in a favorite group that gets automatically updated or something.

**Comment by breversa** [comment]
That would be awesome indeed !

However, please make the auto-update optional (or shedulable), so that OsmAnd doesn't crash or gets slowed down if you have thousands of contacts with addresses.

**Comment by ristein** [comment]
any update here?
searching contacts adresses would be really nice.
an import contacts to favorites might work

**Comment by vshcherb** [comment]
As of today we don't have immediate plans for this

**Comment by gfombell** [comment]
It still is the major missing feature for me.
Like when I check a place I often note: "how this is close to where XYZ leaves! I could meet him aon the same day." Or "I could put those friends of mine who are neighbours in contact with each other."
Note: it would be good to have a parameter to decide from which scale (zoom level) contacts should be shown.
Google Maps shows only from approx 1 cm=500m and in some cases I would like to see on a higher level.
Would it make sense to choose a specific contacts management app / website, the same way sharing ones position was implemented with Telegram?

**Comment by Themanwithoutaplan** [comment]
Not having this feature makes it unnecessarily difficult to route to any contact because of the way the address lookup is handled. From Contacts the address is passed on as Streetname & Number (order depends upon convention), town, postcode, country but OsmAnd likes to search the other way round. This means that you often have to edit the address manually before OsmAnd can find it. Given that travelling to contacts is a fairly frequent occurence, this is a major annoyance.

**Comment by vshcherb** [comment]
>> From Contacts the address is passed on as Streetname & Number (order depends upon convention), town, postcode, country but OsmAnd likes to search the other way roun

It has been changed last release, now it searches by most important keywords first. Since our database of keywords is not complete, sometimes you need to delete some non-meaningful or less important part of address. Though mostly it correctly detects street, city, housenumber. So the order doesn't matter any more

**Comment by Themanwithoutaplan** [comment]
> It has been changed last release, now it searches by most important keywords first. Since our database of keywords is not complete, sometimes you need to delete some non-meaningful or less important part of address. Though mostly it correctly detects street, city, housenumber. So the order doesn't matter any more

Yes, I noticed an improvement but it's still really not good enough, especially when compared with other apps such as Here.

**Comment by stevenroose** [comment]
I think this can be closed as a duplicate of https://github.com/osmandapp/Osmand/issues/3026.

**Comment by vshcherb** [comment]
This is one feels different though it is related

**Comment by Mannshoch** [comment]
As an Idea for how It could look like.

- [ ] Show Pins with a generic Persona Icon (Destination set by Osmand)
- [ ] On tip on a pin you get som basic contact data. 
  - [ ] After tip on the pin you should be able to move the pin on the map.
  - [ ] Osmand should save this info to the specific Contact. 
(I'm not sure how that should look like but [RFC](https://geouri.org/2010/06/10/geo-and-vcards/) from CardDAV looks like `GEO;TYPE=Parking:geo:47.01919,8.36772`)
- [ ] Osmand allow to see unlocatable contacts in a seperate list
- [ ] On tip of sharing a coordinate, send to contact should also be available
- [ ] Contacts that was able to locate by coordinates should have a different icon

**Comment by promeneur** [comment]
Hello

Any news ?

**Comment by promeneur** [comment]
I explain why I want this enhancement.

In my contacts I have many favorites restaurants in my town or other towns. Then, when I need to eat, by just looking at the map around my position I would be able to easily choose one restaurant.

**Comment by vshcherb** [comment]
Quite interesting case actually

**Comment by visika** [comment]
Is this possibly a duplicate of #3026 ?

**Comment by GuiSousa135** [comment]
like this project?

https://github.com/zood/george

**Comment by Themanwithoutaplan** [comment]
> like this project?
> 
> https://github.com/zood/george

No, we've already got OsmAnd Track for that. This is all just about better integration of OsmAnd with the contacts app on the phone. At the moment we have to duplicate the data which is both tiresome and error prone.
